src/ucp/wireup/wireup.c

```diff
@@ -790,28 +799,28 @@ ucp_wireup_connect_lane_to_iface(ucp_ep_h ep, ucp_lane_index_t lane,
         return status;
     }
 
-    if ((ep->uct_eps[lane] == NULL) && ucp_ep_has_cm_lane(ep)) {
-        /* create wireup EP in case of CM lane is used, since a WIREUP EP
-         * is used to keep user's pending requests and send WIREUP MSGs (if
-         * this is WIREUP MSG lane) until CM and WIREUP_MSG phases are done */
-        status = ucp_wireup_ep_create(ep, &ep->uct_eps[lane]);
-        if (status != UCS_OK) {
-            /* coverity[leaked_storage] */
-            return status;
-        }
-    }
-
     if (ep->uct_eps[lane] == NULL) {
-        ucs_trace("ep %p: assign uct_ep[%d]=%p", ep, lane, uct_ep);
-        ep->uct_eps[lane] = uct_ep;
+        if (ucp_ep_has_cm_lane(ep)) {
+            /* create wireup EP in case of CM lane is used, since a WIREUP EP is
+             * used to keep user's pending requests and send WIREUP MSGs (if it
+             * is WIREUP MSG lane) until CM and WIREUP_MSG phases are done */
+            status = ucp_wireup_ep_create(ep, &ep->uct_eps[lane]);
+            if (status != UCS_OK) {
+                /* coverity[leaked_storage] */
+                return status;
+            }
+            ucp_wireup_ep_lane_set_next_ep(ep, lane, uct_ep);
+        } else {
+            ucs_trace("ep %p: assign uct_ep[%d]=%p", ep, lane, uct_ep);
+            ep->uct_eps[lane] = uct_ep;
+        }
     } else {
         /* If EP already exists, it's a wireup proxy, and we need to update
          * its next_ep instead of replacing it.
          */
         ucs_assert(ucp_wireup_ep_test(ep->uct_eps[lane]));
-        ucs_trace("ep %p: wireup uct_ep[%d]=%p next set to %p", ep, lane,
-                  ep->uct_eps[lane], uct_ep);
-        ucp_wireup_ep_set_next_ep(ep->uct_eps[lane], uct_ep);
+        ucs_assert(ucp_proxy_ep_extract(ep->uct_eps[lane]) == NULL);
+        ucp_wireup_ep_lane_set_next_ep(ep, lane, uct_ep);
```

Comment:
in which case the 2iface lane can already exist? old cm flow?

Reply:
added assertion to check that this is possible for old and new sockaddr flows only